Adds the ability to set the default "port" in the rsyncd.conf file.

--- orig/clientserver.c	2004-08-02 02:29:16
+++ clientserver.c	2004-09-28 18:44:33
@@ -598,6 +598,9 @@ int daemon_main(void)
 	if (!lp_load(config_file, 1))
 		exit_cleanup(RERR_SYNTAX);
 
+	if (rsync_port == 0 && (rsync_port = lp_rsync_port()) == 0)
+		rsync_port = RSYNC_PORT;
+
 	log_init();
 
 	rprintf(FLOG, "rsyncd version %s starting, listening on port %d\n",
--- orig/loadparm.c	2004-07-31 16:41:04
+++ loadparm.c	2004-09-28 12:56:51
@@ -104,6 +104,7 @@ typedef struct
 	char *socket_options;
 	int syslog_facility;
 	int max_verbosity;
+	int rsync_port;
 } global;
 
 static global Globals;
@@ -271,6 +272,7 @@ static struct parm_struct parm_table[] =
   {"log file",         P_STRING,  P_GLOBAL, &Globals.log_file,      NULL,  0},
   {"pid file",         P_STRING,  P_GLOBAL, &Globals.pid_file,      NULL,  0},
   {"max verbosity",    P_INTEGER, P_GLOBAL, &Globals.max_verbosity, NULL,  0},
+  {"port",             P_INTEGER, P_GLOBAL, &Globals.rsync_port,    NULL,  0},
 
   {"timeout",          P_INTEGER, P_LOCAL,  &sDefault.timeout,     NULL,  0},
   {"max connections",  P_INTEGER, P_LOCAL,  &sDefault.max_connections,NULL, 0},
@@ -353,6 +355,7 @@ FN_GLOBAL_STRING(lp_pid_file, &Globals.p
 FN_GLOBAL_STRING(lp_socket_options, &Globals.socket_options)
 FN_GLOBAL_INTEGER(lp_syslog_facility, &Globals.syslog_facility)
 FN_GLOBAL_INTEGER(lp_max_verbosity, &Globals.max_verbosity)
+FN_GLOBAL_INTEGER(lp_rsync_port, &Globals.rsync_port)
 
 FN_LOCAL_STRING(lp_name, name)
 FN_LOCAL_STRING(lp_comment, comment)
--- orig/options.c	2004-09-23 17:42:07
+++ options.c	2004-09-28 18:43:43
@@ -127,7 +127,7 @@ char *password_file = NULL;
 char *rsync_path = RSYNC_PATH;
 char *backup_dir = NULL;
 char backup_dir_buf[MAXPATHLEN];
-int rsync_port = RSYNC_PORT;
+int rsync_port = 0;
 int link_dest = 0;
 
 int verbose = 0;
--- orig/rsyncd.conf.yo	2004-09-22 02:02:04
+++ rsyncd.conf.yo	2004-09-28 18:53:35
@@ -160,6 +160,10 @@ being downloaded/uploaded by adding an e
 is a relatively new feature in rsync, so make sure your server is running
 at least 2.6.3 to effect this.
 
+dit(bf(port)) You can override the default port the daemon will listen on
+by specifying this value (defaults to 873).  This is ignored if the daemon
+is being run by inetd, and is superseded by the --port command-line option.
+
 dit(bf(max connections)) The "max connections" option allows you to
 specify the maximum number of simultaneous connections you will allow.
 Any clients connecting when the maximum has been reached will receive a
